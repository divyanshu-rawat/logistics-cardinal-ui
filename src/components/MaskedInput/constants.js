import createNumberMask from 'text-mask-addons/dist/createNumberMask';
import emailMask from 'text-mask-addons/dist/emailMask';

// DD.MM.YYYY
export const BIRTHDAY_MASK = [
  /\d/,
  /\d/,
  '.',
  /\d/,
  /\d/,
  '.',
  /\d/,
  /\d/,
  /\d/,
  /\d/,
];

// HH:mm
export const TIME_MASK = [/[0-9]/, /[0-9]/, ':', /[0-9]/, /[0-9]/];

// 10 % || - 10 %
export const PERCENTAGE_MASK = createNumberMask({
  prefix: '',
  includeThousandsSeparator: false,
  suffix: ' %',
  allowNegative: true,
});

// $ 10
export const MONEY_MASK = createNumberMask();

// rick@c137.com
export const EMAIL_MASK = emailMask;

// 10.29
export const DECIMAL_MASK = createNumberMask({
  prefix: '',
  allowDecimal: true,
  requireDecimal: true,
});

export default {
  BIRTHDAY_MASK,
  MONEY_MASK,
  EMAIL_MASK,
  DECIMAL_MASK,
  PERCENTAGE_MASK,
};
