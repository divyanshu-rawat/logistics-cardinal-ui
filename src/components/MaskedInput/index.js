import React from 'react';
import PropTypes from 'prop-types';
import MaskedInput from 'react-text-mask';

import Input from '../Input';

const InputMask = ({
  guide,
  mask,
  placeholder,
  id,
  value,
  onChange,
  onBlur,
}) => (
  <MaskedInput
    mask={mask}
    guide={guide}
    placeholder={placeholder}
    value={value}
    id={id}
    onChange={onChange}
    onBlur={onBlur}
    render={(ref, props) => <Input inputRef={ref} {...props} />}
  />
);

InputMask.propTypes = {
  /** Mask can be Array or Func. Check `react-text-mask` for more info about it */
  mask: PropTypes.oneOfType([PropTypes.array, PropTypes.func]).isRequired,
  /** input id */
  id: PropTypes.string.isRequired,
  /** onChange callback, if you are using redux-form it's going likely to be `input.change` */
  onChange: PropTypes.func.isRequired,
  /** input value */
  value: PropTypes.any,
  /** onBlur callback */
  onBlur: PropTypes.func,
  /** input placeholder */
  placeholder: PropTypes.string,
  /** Set if mask will guide the user or not. */
  guide: PropTypes.bool,
};

export default InputMask;
