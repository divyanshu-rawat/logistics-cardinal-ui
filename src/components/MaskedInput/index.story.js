import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';
import { select } from '@storybook/addon-knobs';

import withTests from '../../utils/withTests';

import { FORM } from '../../../.storybook/sections';

import MaskedInput from '.';
import {
  BIRTHDAY_MASK,
  PERGENTAGE_MASK,
  MONEY_MASK,
  EMAIL_MASK,
} from './constants';

const maskOptions = ['birthday', 'percentage', 'currency', 'email'];
const masks = {
  birthday: {
    id: 'birthday',
    mask: BIRTHDAY_MASK,
    placeholder: 'DD.MM.YYYY',
    guide: true,
  },
  percentage: {
    id: 'percentage',
    mask: PERGENTAGE_MASK,
    placeholder: '%',
    guide: false,
  },
  currency: {
    id: 'currency',
    mask: MONEY_MASK,
    placeholder: '$',
    guide: false,
  },
  email: {
    id: 'email',
    mask: EMAIL_MASK,
    placeholder: 'rick@c137.com',
    guide: true,
  },
};

storiesOf(`${FORM}MaskedInput`, module)
  .addDecorator(withTests('MaskedInput/tests/index'))
  .add(
    'Simple',
    withInfo()(() => {
      const selectedOption = select('select mask', maskOptions, 'birthday');

      return (
        <MaskedInput
          mask={masks[selectedOption].mask}
          id={masks[selectedOption].id}
          guide={masks[selectedOption].guide}
          onChange={(data) => console.log(data)}
          placeholder={masks[selectedOption].placeholder}
        />
      );
    }),
  );
