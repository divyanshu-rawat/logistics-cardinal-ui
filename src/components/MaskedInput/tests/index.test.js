import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ThemeProvider } from 'styled-components';

import MaskedInput from '..';

import Theme from '../../../themes/rooster';

import {
  BIRTHDAY_MASK,
  PERGENTAGE_MASK,
  MONEY_MASK,
  EMAIL_MASK,
} from '../constants';

const masks = {
  birthday: {
    id: 'birthday',
    mask: BIRTHDAY_MASK,
    placeholder: 'DD.MM.YYYY',
    guide: true,
  },
  percentage: {
    id: 'percentage',
    mask: PERGENTAGE_MASK,
    placeholder: '%',
    guide: false,
  },
  currency: {
    id: 'currency',
    mask: MONEY_MASK,
    placeholder: '$',
    guide: false,
  },
  email: {
    id: 'email',
    mask: EMAIL_MASK,
    placeholder: 'rick@c137.com',
    guide: true,
  },
};

const onChangeFn = jest.fn();
const onBlurFn = jest.fn();

const renderedComponent = ({ mask, id, placeholder, guide }) =>
  mount(
    <ThemeProvider theme={Theme}>
      <MaskedInput
        mask={mask}
        guide={guide}
        placeholder={placeholder}
        id={id}
        onChange={onChangeFn}
        onBlur={onBlurFn}
      />
    </ThemeProvider>,
  );

describe('<MaskedInput />', () => {
  it('should render the component with `birthday mask`', () => {
    const tree = toJson(renderedComponent(masks.birthday));
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with `percentage mask`', () => {
    const tree = toJson(renderedComponent(masks.percentage));
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with `currency mask`', () => {
    const tree = toJson(renderedComponent(masks.currency));
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with `email mask`', () => {
    const tree = toJson(renderedComponent(masks.currency));
    expect(tree).toMatchSnapshot();
  });

  it('MaskedInput: should call `onChangeFn` when value is changed', () => {
    const component = renderedComponent(masks.currency);
    component
      .find('input')
      .simulate('change', { event: { target: { value: 123 } } });

    expect(onChangeFn.mock.calls.length).toBe(1);
  });

  it('MaskedInput: should call `onBlurFn` when value is changed', () => {
    const component = renderedComponent(masks.currency);

    component.find('input').simulate('blur');

    expect(onBlurFn.mock.calls.length).toBe(1);
  });
});
