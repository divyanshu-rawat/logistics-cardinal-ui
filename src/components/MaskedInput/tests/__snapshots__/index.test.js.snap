// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaskedInput /> should render the component with \`birthday mask\` 1`] = `
<InputMask
  guide={true}
  id="birthday"
  mask={
    Array [
      /\\\\d/,
      /\\\\d/,
      ".",
      /\\\\d/,
      /\\\\d/,
      ".",
      /\\\\d/,
      /\\\\d/,
      /\\\\d/,
      /\\\\d/,
    ]
  }
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="DD.MM.YYYY"
>
  <t
    guide={true}
    id="birthday"
    mask={
      Array [
        /\\\\d/,
        /\\\\d/,
        ".",
        /\\\\d/,
        /\\\\d/,
        ".",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
      ]
    }
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="DD.MM.YYYY"
    render={[Function]}
  >
    <Input
      id="birthday"
      inputRef={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="DD.MM.YYYY"
    >
      <input
        className="form-control"
        id="birthday"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="DD.MM.YYYY"
      />
    </Input>
  </t>
</InputMask>
`;

exports[`<MaskedInput /> should render the component with \`currency mask\` 1`] = `
<InputMask
  guide={false}
  id="currency"
  mask={[Function]}
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="$"
>
  <t
    guide={false}
    id="currency"
    mask={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="$"
    render={[Function]}
  >
    <Input
      id="currency"
      inputRef={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="$"
    >
      <input
        className="form-control"
        id="currency"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="$"
      />
    </Input>
  </t>
</InputMask>
`;

exports[`<MaskedInput /> should render the component with \`email mask\` 1`] = `
<InputMask
  guide={false}
  id="currency"
  mask={[Function]}
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="$"
>
  <t
    guide={false}
    id="currency"
    mask={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="$"
    render={[Function]}
  >
    <Input
      id="currency"
      inputRef={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="$"
    >
      <input
        className="form-control"
        id="currency"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="$"
      />
    </Input>
  </t>
</InputMask>
`;

exports[`<MaskedInput /> should render the component with \`percentage mask\` 1`] = `
<InputMask
  guide={false}
  id="percentage"
  mask={[Function]}
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="%"
>
  <t
    guide={false}
    id="percentage"
    mask={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="%"
    render={[Function]}
  >
    <Input
      id="percentage"
      inputRef={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="%"
    >
      <input
        className="form-control"
        id="percentage"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="%"
      />
    </Input>
  </t>
</InputMask>
`;
